inference-engine/src/transformations/src/transformations/common_optimizations/add_fake_quantize_fusion.cpp

```diff
@@ -29,38 +30,83 @@ ngraph::pass::AddFakeQuantizeFusion::AddFakeQuantizeFusion() {
+            is_single_value = op::util::get_single_value(add_const, v);
+            if (is_single_value) {
+                new_const = std::make_shared<opset5::Constant>(add_const->get_element_type(), Shape{1}, v);
+            }
+        }
+
+        if (!is_single_value) {
             // disallow constant shapes other than (N, 1, 1, ..., 1) or (1, C, 1, ..., 1)
             if (!(const_shape[0] > 1 && const_shape[0] == const_shape_size) &&
                 !(const_shape.size() > 1 && const_shape[1] == const_shape_size)) {
                 return false;
             }
-        }
 
-        if (const_shape_size > 1 &&
-            static_cast<Dimension::value_type>(const_shape.size()) < fq->get_input_partial_shape(0).rank().get_length()) {
-            // Reshape constants like (C, 1, 1) to (1, C, 1, 1)
-            const_shape.insert(const_shape.begin(), fq->get_input_partial_shape(0).rank().get_length() - const_shape.size(), 1);
-            add_const = std::make_shared<opset5::Reshape>(add_const, op::Constant::create(element::u64, Shape{const_shape.size()}, const_shape), false);
+            const auto& add_inputs = add_node->input_values();
+            bool add_parent_is_conv_or_mm = std::any_of(add_inputs.begin(), add_inputs.end(),
+                                                        [] (const Output<Node>& node) -> bool {
+                                                            auto node_ptr = node.get_node();
+                                                            return is_type<opset5::Convolution>(node_ptr) ||
+                                                                   is_type<opset5::GroupConvolution>(node_ptr) ||
+                                                                   is_type<opset5::ConvolutionBackpropData>(node_ptr) ||
+                                                                   is_type<opset5::GroupConvolutionBackpropData>(node_ptr) ||
+                                                                   is_type<opset5::MatMul>(node_ptr);
+                                                        });
+            if (add_parent_is_conv_or_mm)
+                return false;
+            auto fq_users = fq->get_users();
+            // Concat LPT transformation supports per tensor quantization only
+            bool fq_user_is_concat = std::any_of(fq_users.begin(), fq_users.end(),
+                                                 [] (const Output<Node>& node) -> bool {
+                                                     auto node_ptr = node.get_node();
+                                                     return is_type<opset5::Concat>(node_ptr);
+                                                 });
+            if (fq_user_is_concat)
+                return false;
+            auto diff = fq->get_input_partial_shape(0).rank().get_length() - static_cast<Dimension::value_type>(const_shape.size());
+            if (diff > 0) {
+                // Reshape constants like (C, 1, 1) to (1, C, 1, 1)
```

Comment:
Should we check the opposite case when Constant has `[1,C,1,1]` shape and FQ input has `[C,1,1]`?

Reply:
I don't think we can have this situation in the wild. If Constant is 4D, then the Add output is 4D and so is FQ input.